### PR TITLE
Update ZAP to force lists to be external-storage.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
The Power Source cluster addition is because that was one of two lists
that we did not already have AttributeAccessInterface handling for
(and the other one was in a cluster that already had an
AttributeAccessInterface).

Now that we have AttributeAccessInterface for all lists, we no longer
need the fake "return an empty list" thing in
ember-compatibility-functions.cpp.

#### Problem
Various lists are not marked as external in ZAP config.

#### Change overview
Force them to be treated as external automatically no matter what the UI says.  Implement a Power Source cluster to actually provide one of the lists tests try to read.  Fix various other clusters' attribute access interfaces to output empty lists as needed instead of just falling back to the attribute store, since these things no longer live in the attribute store.

#### Testing
Should be no observable behavior changes, apart from less RAM use.

### CODE REVIEW

File: src/app/clusters/channel-server/channel-server.cpp [hunk 1]
```diff
@@ -122,20 +122,30 @@ CHIP_ERROR ChannelAttrAccess::Read(const app::ConcreteReadAttributePath & aPath,
     EndpointId endpoint = aPath.mEndpointId;
     Delegate * delegate = GetDelegate(endpoint);
 
-    if (isDelegateNull(delegate, endpoint))
```
Comment (jmeg-sfy) [inline src/app/clusters/channel-server/channel-server.cpp:125]: questiom : there is no method to say isList ?   and then launch aEncoder.EncodeEmptyList();

File: src/app/clusters/power-source-server/power-source-server.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::PowerSource::Attributes;
+
+namespace {
+
+class PowerSourceAttrAccess : public AttributeAccessInterface
+{
```
Comment (jmeg-sfy) [inline src/app/clusters/power-source-server/power-source-server.cpp:38]: reading the title we don't know that you also added this missing cluster server..

**Review by jmeg-sfy — APPROVED** [review]

File: src/app/clusters/channel-server/channel-server.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/clusters/channel-server/channel-server.cpp:125]: There is, sort of: you could query the metadata and then examine its type.

But fundamentally this code is just buggy: I would think that no delegate would mean you return an error!  For now I'm just preserving the existing behavior, so tests will keep passing, but that behavior is wrong....

File: src/app/clusters/power-source-server/power-source-server.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/clusters/power-source-server/power-source-server.cpp:38]: Good point.  I wrote the title and commit message before I realized I had to add this to pass tests....

Comment (bzbarsky-apple) [reply src/app/clusters/power-source-server/power-source-server.cpp:38]: Updated PR description to make things clearer.

**Review by tcarmelveilleux — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: change by domain owner, has been up sufficiently for cross timezone review.

**Pull request merged**